query-algebrizer/src/clauses/mod.rs

```diff
@@ -283,35 +284,41 @@ impl ConjoiningClauses {
         }
     }
 
-    pub fn bind_column_to_var(&mut self, schema: &Schema, table: TableAlias, column: DatomsColumn, var: Variable) {
+    pub fn bind_column_to_var<C: Into<Column>>(&mut self, schema: &Schema, table: TableAlias, column: C, var: Variable) {
+        let column = column.into();
         // Do we have an external binding for this?
         if let Some(bound_val) = self.bound_value(&var) {
             // Great! Use that instead.
             // We expect callers to do things like bind keywords here; we need to translate these
             // before they hit our constraints.
+            // We shouldn't need to handle non-fixed columns here: they'll eventually be bound
+            // down in some nested pattern to a value or non-value datoms column.
             // TODO: recognize when the valueType might be a ref and also translate entids there.
-            if column == DatomsColumn::Value {
-                self.constrain_column_to_constant(table, column, bound_val);
-            } else {
-                match bound_val {
-                    TypedValue::Keyword(ref kw) => {
-                        if let Some(entid) = self.entid_for_ident(schema, kw) {
+            match column {
+                Column::Fixed(DatomsColumn::Value) | Column::Variable(_) => {
+                    self.constrain_column_to_constant(table, column, bound_val);
+                }
+                _ => {
```

Comment:
In general, I've found wildcard matches to be worse than listing all the possibilities.  It's really nice to have the compiler tell you where you need to handle a newly added enum tag.